warn/warn_bazel_api.go

```diff
@@ -610,3 +610,58 @@ func ruleImplReturnWarning(f *build.File, fix bool) []*Finding {
 
 	return findings
 }
+
+var legacyNamedParameters = map[string]string {
+	"all": "elements",
+	"any": "elements",
+	"tuple": "x",
+	"list": "x",
+	"len": "x",
+	"str": "x",
+	"repr": "x",
+	"bool": "x",
+	"int": "x",
+	"dir": "x",
+	"type": "x",
+	"hasattr": "x",
+	"getattr": "x",
+	"select": "x",
```

Comment:
Most often I see string.split(maxsplit=-1) or dict.get(key, default=...).  Be sure to handle methods too.

Reply:
I've collected the potential cases here: https://github.com/bazelbuild/bazel/issues/5010#issuecomment-466472598

I'll implement more fixes once we agree on which keyword parameters we want like to abandon.

Reply:
To repeat what I said on that issue, the important distinction is between core Starlark (e.g. getattr, splitlines) and the build language (e.g. fail, aspect). In the core, following Python, very few functions accept named arguments; int(base=...) is one of the rare exceptions. In the build language, and for user-defined macros, it's much more common for functions to accept or even require named arguments. Let me know if you disagree.